cmd/deploy/deploy.go

```diff
@@ -258,6 +262,16 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, deployOption
 			}
 		}
 
+		if deployOptions.Dependencies && okteto.IsOkteto() {
```

Comment:
if `dependencies`is defined in the manifest, and `!okteto.IsOkteto()`, we should fail when reading the manifest

Also, we should deploy the dependencies on `okteto deploy` if they weren't not deployed even if `--dependencies is not specified in the deploy command